effekt/jvm/src/main/scala/effekt/Server.scala

```diff
@@ -505,10 +505,14 @@ class Server(config: EffektConfig, compileOnChange: Boolean=false) extends Langu
     // nested under an "effekt" key `{ "effekt": { "showIR": "core", ... } }`
     // The former is sent by the VSCode extension for `initializationOptions`,
     // the latter by newer extension versions for `workspace/didChangeConfiguration`.
-    val newSettings = params.getSettings.asInstanceOf[JsonElement].getAsJsonObject
-    this.settings = newSettings;
-    if (newSettings == null) return
-    val effektSection = newSettings.get("effekt")
+    val newSettings = params.getSettings.asInstanceOf[JsonElement]
+    // When the settings come via `initializationOptions`, they can be null as per the LSP spec.
+    if (newSettings.isJsonNull) {
+      return;
+    }
```

Comment:
Shouldn't this be semantically something more like "if the json is null, just pretend it's an empty object"?
Because now it seems like if the options change to `null`, we just ignore the change?